test_tools/sigma_verify_rules.py

```diff
@@ -0,0 +1,170 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""A tool to test sigma rules.
+This tool can be used to verify your rules before running an analyzer.
+It also does not require you to have a full blown Timesketch instance up and running.
+Example way of running the tool:
+  $ sigma_verify_rules.py --config_file ../data/sigma_config.yaml ../data/linux/
+"""
+
+
+import logging
+import os
+import argparse
+import sys
+
+
+
+from sigma.backends import elasticsearch as sigma_elasticsearch
+import sigma.configuration as sigma_configuration
+from sigma.parser import collection as sigma_collection
+logging.basicConfig(level=os.environ.get("LOGLEVEL", "ERROR"))
+import sigma.parser.exceptions
+
+
+def get_codepath():
+    """Return the absolute path to where the tool is run from."""
+    path = __file__
+    if path.startswith(os.path.sep):
+        return path
+
+    dirname = os.path.dirname(path)
+    for sys_path in sys.path:
+        if sys_path.endswith(dirname):
+            return sys_path
+    return dirname
+
+
+def run_verifier(rules_path, config_file_path):
+    """Run an sigma parsing test on a given dir and returns results from the run.
+    Args:
+        rules_path: the path to the rules.
+        config_file_path: the path to a config file that contains mapping data.
+    Raises:
+        IOError: if the path to either test or analyzer file does not exist
+                 or if the analyzer module or class cannot be loaded.
+    Returns:
+        two lists:
+            - sigma_verified_rules with rules that can be added
+            - sigma_rules_with_problems with rules that should not be added
+    """
+    if not os.path.isdir(rules_path):
+        raise IOError('Rules not found at path: {0:s}'.format(
+            rules_path))
+    if not os.path.isfile(config_file_path):
+        raise IOError('Config file path not found at path: {0:s}'.format(
+            config_file_path))
+
+    sigma_config_path = config_file_path
+
+    with open(sigma_config_path, 'r') as sigma_config_file:
+        sigma_config_con = sigma_config_file.read()
+    sigma_config = sigma_configuration.SigmaConfiguration(sigma_config_con)
+    sigma_backend = sigma_elasticsearch.ElasticsearchQuerystringBackend(sigma_config, {})
+
+    for dirpath, dirnames, files in os.walk(rules_path):
+
+        if 'deprecated' in dirnames:
+            dirnames.remove('deprecated')
+
+        rule_extensions = ("yml","yaml")
+
+        sigma_verified_rules  = []
+        sigma_rules_with_problems = []
+
+        for rule_filename in files:
+            if rule_filename.lower().endswith(rule_extensions):
+
+                # if a sub dir is found, append it to be scanned for rules
+                if os.path.isdir(os.path.join(rules_path, rule_filename)):
+                    logging.error(
```

Comment:
is this really an error?

also start with a capitol letter (s/this/This/)

Reply:
True